app/services/subscriptions_reminder.rb

```diff
@@ -9,25 +9,32 @@ def generate
 
   private
 
-  # Users who 
+  # Users who
   #   haven't been reminded in last X hours
   #   have unread messages
   #   are donors with active offers
   #   aren't the author of the message
+  #   its not a private messages
+  #   its not order related messages
+  #   are donors with other roles such as reviewer will receive sms on offers where they are acting as \
+  #   donor
   # If sms_reminder_sent_at is NULL then use created_at so we don't SMS user immediately
   def user_candidates_for_reminder
     states = ['submitted', 'under_review', 'reviewed', 'scheduled', 'received',
       'receiving', 'inactive'] # NOT draft, closed or cancelled
     user_ids = Offer.where(state: states).distinct.pluck(:created_by_id)
-    User.joins(subscriptions: [:message]).
+    User.joins(subscriptions: [:message, :offer]).
       where('users.id IN (?)', user_ids).
       where("COALESCE(users.sms_reminder_sent_at, users.created_at) < (?)", delta.iso8601).
       where('subscriptions.state': 'unread').
       where("messages.created_at > COALESCE(users.sms_reminder_sent_at, users.created_at)").
+      where("messages.offer_id IS NOT NULL OR messages.item_id IS NOT NULL and messages.order_id IS NULL").
```

Comment:
only `order_id` null can work but added extra checks and I don't see any harm in it. :)